[ph-nathan] iP

ph-nathan opened the pull request:
# Pippi
> Also known as Clefairy, the intended Pokémon mascot. #justiceforclefairy

As a trustworthy partner, Pippi frees your mind from amnesia of having to remember tasks you need to do. Pippi is: 
- Pink 
- Bulky
- Super fast (has a speed base of 35, but knows Trick Room so can outspeed any opponent)
- known for having an expansive movepool, great typing and top tier abilities
- **Super cute**

You can entrust Pippi with remembering tasks by
1. Going to [Mt.Moon](https://bulbapedia.bulbagarden.net/wiki/Mt._Moon) 
2. Catch a clefairy
3. Befriend them
4. Tell them your tasks
5. Return them to pokeball once finished 😿 

Features:
- [x] Managing tasks 
- [ ] Reminders (coming soon)
- [ ] Telling random fact about Pippi (coming soon)

- - - 
If you Java programmer, you can ask Pippi for help practicing Java too. Here's the main method:
```java
public class Pippi {
    public static void main(String[] args) {
        Pippi pippi = new Pippi();
        pippi.start();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ph-nathan committed:
Refactor code for level 0: Rename, Greet and Exit

ph-nathan committed:
Add logic for Level 1: Echo

ph-nathan committed:
Add logic for adding tasks and listing for Level 2: Add, list

ph-nathan committed:
Add logic for marking and unmarking of tasks for Level 3: Mark as done

ph-nathan committed:
Add logic to handle adding different event types for Level 4: ToDo, Event, Deadline

ph-nathan committed:
Add testing for Level-A-TextUiTesting

ph-nathan committed:
Add exception handling for Level-5: Handle Error

ph-nathan committed:
Fix bug

ph-nathan committed:
Fix error with previous commit and add logic for deleting events for Level 5 + 6

ph-nathan committed:
Add logic for saving and retrieving saved tasks for Level 7: Save

ph-nathan committed:
Merge branch 'branch-Level-7' into master branch

ph-nathan committed:
Add date formatter for deadline tasks for Level-8

ph-nathan committed:
Merge branch 'branch-Level-8'

ph-nathan committed:
Refactor codebase to follow OOP for increment A

ph-nathan committed:
Merge branch 'branch-A-MoreOOP'

ph-nathan committed:
Refactor code into packages for increment A-Packages

ph-nathan committed:
Merge branch 'branch-A-Packages'

ph-nathan committed:
Merge branch 'add-gradle-support'

ph-nathan committed:
Add gradle to project for increment A-Gradle

ph-nathan committed:
Add JUnit tests for Task and DateFormatter classes

ph-nathan committed:
Merge branch 'branch-A-JUnit'

ph-nathan committed:
Add logic for finding tasks for Level-9: Find

ph-nathan committed:
Merge branch 'branch-Level-9'

ph-nathan committed:
Add javadoc for methods for increment A-JavaDoc

ph-nathan committed:
Merge branch 'branch-A-JavaDoc'

howenc commented on src/main/java/pippi/storage/Storage.java:
```diff
@@ -0,0 +1,93 @@
+import java.io.IOException;
+import java.util.ArrayList;
+public class Storage {
+    private static final String filePath = "./data/Pippi.txt";
```
Since this is static and final, it should be considered a constant right?
Should you capitalize it?
<img width="397" alt="image" src="https://github.com/nus-cs2103-AY2324S1/ip/assets/107026005/fbb92ad0-cf22-4b3a-9a4b-e23be55c728d">

howenc commented on src/main/java/pippi/storage/Storage.java:
```diff
@@ -0,0 +1,93 @@
+                        tasks.add(d);
+                        break;
+                    case "E":
+                        String start = parts[3].split("to ")[0];
```
While I like the simplicity in expression and the fact that your code is easy to read, I forgot what `parts[3]` was.
Should there be a better name here?

howenc commented on src/main/java/pippi/parser/Parser.java:
Overall this file is very easy to read.
The Parser through the reply method will automatically parse through the input given.
It will then add the task to the task list.

howenc commented on src/main/java/pippi/parser/Parser.java:
```diff
@@ -0,0 +1,124 @@
+            String[] input = userMessage.split(" ", 2);
+            String command = input[0];
+            switch (command) {
+                case "bye":
```
There is no indent between the switch and case.
<img width="397" alt="image" src="https://github.com/nus-cs2103-AY2324S1/ip/assets/107026005/b40030d1-a5c5-4b3e-bb01-55d33100a5a9">

howenc commented on src/main/java/pippi/parser/Parser.java:
```diff
@@ -0,0 +1,124 @@
+                    break;
+                case "deadline":
+                    if (input.length < 2 || input[1].trim().isEmpty()) {
+                        throw new PippiException("Metronome!!! The description and due time of a deadline cannot be empty.");
```
Is this exceeding the character limit?
<img width="392" alt="image" src="https://github.com/nus-cs2103-AY2324S1/ip/assets/107026005/2b802694-02df-4e74-9f5d-f7eef1a223aa">

howenc commented on src/main/java/pippi/parser/Parser.java:
```diff
@@ -0,0 +1,124 @@
+                    break;
+                case "event":
+                    if (input.length < 2 || input[1].trim().isEmpty()) {
+                        throw new PippiException("Pound!!! The description and start, end time of an event cannot be empty.");
```
Similar to the previous comment, is this exceeding the character limit?

howenc commented on src/main/java/pippi/parser/DateFormatter.java:
```diff
@@ -0,0 +1,41 @@
+import java.time.format.DateTimeParseException;
+
+public class DateFormatter {
+    public static final DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
Since this is static and final, it should be considered a constant right?
Should you capitalize it?
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/107026005/e7564b65-1bf6-4aba-96a4-91091b5e2438)

howenc commented on src/main/java/pippi/parser/DateFormatter.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    public static boolean isValidFormat(String date) {
+        try {
+            System.out.println("here");
```
Um, I think you left this here by accident.

howenc commented on src/main/java/pippi/task/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+
+    /**
+     * Returns the String representation of a deadline task to the UI
+     * @return Deadline string representation
```
I think there should be a blank line between description and parameter?
I didn't comment this on the rest as it would be too much.
<img width="394" alt="image" src="https://github.com/nus-cs2103-AY2324S1/ip/assets/107026005/a6512401-c842-4398-ba30-199ec6480e55">

howenc commented on src/test/java/pippi/PippiTest.java:
```diff
@@ -0,0 +1,17 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PippiTest {
```
Um, I think you forgot some things here.

howenc reviewed (changes requested):
No major problems, just come code style changes needed.
All of your code is very easy to read. 
I like the flow of logic and they way you name things. 
The all the names became more intuitive as I read more of your code.
However, if I'm just reading clips of your code, the names would not give me much context.

lyuanww commented on src/main/java/pippi/parser/Parser.java:
The `reply` method is quite long.
Perhaps a shorter method by abstracting some parts of code out?

lyuanww commented on src/main/java/pippi/task/Task.java:
```diff
@@ -0,0 +1,63 @@
+        this.isDone = false;
+    }
+
+    public String toMemory() {
```
The naming of the `toMemory` method is quite confusing.
Maybe naming the method so that it is clear that it is returning a String formatted to save in the local file.

lyuanww commented on src/main/java/pippi/Pippi.java:
```diff
@@ -0,0 +1,34 @@
+        Scanner scanner = new Scanner(System.in);
+
+        while (!this.inPokeball) {
+            String userMessage = scanner.nextLine();
```
Instead of `userMessage`, maybe `userInput` will be more suitable?

lyuanww commented on src/main/java/pippi/storage/Storage.java:
```diff
@@ -0,0 +1,93 @@
+     * (array is empty if file is empty or not found)
+     * @return Array List of tasks
+     */
+    public static ArrayList<Task> read() {
```
Maybe a more meaningful `readTasks`?

lyuanww reviewed (commented):
Overall, good job! Just a few nits to fix.

ph-nathan committed:
Implement GUI for Pippi

ph-nathan committed:
Enhance GUI for Level-10

ph-nathan committed:
Merge branch 'branch-Level-10'

ph-nathan committed:
Configure CheckStyle functionality for A-CheckStyle

ph-nathan committed:
Merge branch 'branch-A-CheckStyle'

ph-nathan committed:
Fix minor bug

ph-nathan committed:
Add assertions

Some private methods and cases of switch statement hasn't been handled correctly.
Private arguments are provided by the developer’s code only and the developer may
want to check his/her assumptions about arguments.

Let's add assertion statements to methods such as handleUserInput() and reply().

ph-nathan committed:
Merge pull request #2 from ph-nathan/branch-A-Assertions

Add assertions

ph-nathan committed:
Add javadocs and change some namings to adhere to code quality

Some javadocs for public classes and methods were missing

Let's add documentation to them, and rename some methods to follow more closely the naming quality and conventions

ph-nathan committed:
Merge pull request #3 from ph-nathan/branch-A-CodeQuality

Add javadocs and change some namings to adhere to code quality

ph-nathan committed:
Add C-Help extension

ph-nathan committed:
Merge pull request #4 from ph-nathan/branch-BCD-Extension

Add C-Help extension

ph-nathan committed:
Refine GUI and Personality

ph-nathan committed:
Update user guide

ph-nathan committed:
Fix user guide

ph-nathan committed:
Update storage

ph-nathan committed:
Fix storage

ph-nathan committed:
Merge pull request #5 from ph-nathan/branch-A-BetterGUI

Refine GUI and Personality

ph-nathan committed:
Fix style and bug for jar file
